adi/context_manager.py

```diff
@@ -36,3 +36,19 @@ def __init__(self, uri="", _device_name=""):
                 self._ctx = iio.Context(self.uri)
         except BaseException:
             raise Exception("No device found")
+
+    def close(self):
+        """Close the IIO context"""
+        if self._ctx:
+            del self._ctx
+            self._ctx = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+        if exc_type:
+            print(f"Exception caught: {exc_value}")
+            return True
+        return False
```

Comment:
You probably just want to replace these with "return False" so exceptions propagate normally. Library modules shouldn't print like this.